src/identities.rs

```diff
@@ -79,6 +79,14 @@ pub trait One: Sized + Mul<Self, Output = Self> {
     /// `static mut`s.
     // FIXME (#5527): This should be an associated constant
     fn one() -> Self;
+
+    /// Returns `true` if `self` is equal to the multiplicative identity.
+    ///
+    /// Compatibility note: this method will be a requirement to implement in the future; new
+    /// implementors should provide it for future-compatibility.
```

Comment:
I'm a little hesitant about this scary-sounding note, because we're definitely *not* going to change that without a semver bump!  And when we do, it should also drop the `Self: PartialEq`, which means they'll have porting work to do anyway.  Maybe instead just advise that implementations should consider adding it now for better performance, noting that it will only become required in a future breaking change.

Also, let's put `#[inline]` on this.